contracts/deploy/Create2Deployer.sol

```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import { DeployerBase } from './DeployerBase.sol';
+import { DeployCreate2 } from './DeployCreate2.sol';
+
+/**
+ * @title Create2Deployer Contract
+ * @notice This contract is responsible for deploying and initializing new contracts using the `CREATE2` method
+ * which computes the deployed contract address based on the bytecode, deployer address, and deployment salt.
+ */
+contract Create2Deployer is DeployCreate2, DeployerBase {
+    function _deploy(bytes memory bytecode, bytes32 deploySalt) internal override returns (address) {
```

Comment:
```suggestion
    function _deploy(bytes memory bytecode, bytes32 salt) internal override returns (address) {
```
-> `salt` everywhere? it being for deployment is implied i think

Reply:
salt and deploySalt are different terms, salt is from the user, deploySalt is hashed with `msg.sender`

Reply:
yeah, i feel like it could be named the other way around too. but fine with keeping it separate